pkg/vpcmodel/semanticDiff.go

```diff
@@ -29,64 +36,100 @@ type connectionDiff struct {
 	diff  DiffType
 }
 
-type SubnetsDiff map[VPCResourceIntf]map[VPCResourceIntf]*connectionDiff
+type connectivesDiff map[VPCResourceIntf]map[VPCResourceIntf]*connectionDiff
 
 type ConfigsForDiff struct {
-	config1 *VPCConfig
-	config2 *VPCConfig
+	config1      *VPCConfig
+	config2      *VPCConfig
+	diffAnalysis diffAnalysisType
 }
 
-type SubnetConfigConnectivity struct {
-	config             *VPCConfig
-	subnetConnectivity SubnetConnectivityMap
+type configConnectivity struct {
+	config       *VPCConfig
+	connectivity generalConnectivityMap
 }
 
-type DiffBetweenSubnets struct {
-	subnet1Subtract2 SubnetsDiff
-	subnet2Subtract1 SubnetsDiff
+type diffBetweenCfgs struct {
+	cfg1ConnRemovedFrom2 connectivesDiff
+	cfg2ConnRemovedFrom1 connectivesDiff
+	diffAnalysis         diffAnalysisType
 }
 
-func (configs ConfigsForDiff) GetSubnetsDiff(grouping bool) (*DiffBetweenSubnets, error) {
-	// 1. compute connectivity for each of the subnets
-	subnetsConn1, err := configs.config1.GetSubnetsConnectivity(true, grouping)
-	if err != nil {
-		return nil, err
-	}
-	subnetsConn2, err := configs.config2.GetSubnetsConnectivity(true, grouping)
-	if err != nil {
-		return nil, err
+func (configs ConfigsForDiff) GetDiff(grouping bool) (*diffBetweenCfgs, error) {
+	// 1. compute connectivity for each of the configurations
+	var generalConnectivityMap1, generalConnectivityMap2 generalConnectivityMap
+	if configs.diffAnalysis == Subnets {
+		subnetsConn1, err := configs.config1.GetSubnetsConnectivity(true, grouping)
+		if err != nil {
+			return nil, err
+		}
+		subnetsConn2, err := configs.config2.GetSubnetsConnectivity(true, grouping)
+		if err != nil {
+			return nil, err
+		}
+		generalConnectivityMap1 = subnetsConn1.AllowedConnsCombined
+		generalConnectivityMap2 = subnetsConn2.AllowedConnsCombined
+	} else if configs.diffAnalysis == Vsis {
+		connectivity1, err := configs.config1.GetVPCNetworkConnectivity(grouping)
+		if err != nil {
+			return nil, err
+		}
+		connectivity2, err := configs.config2.GetVPCNetworkConnectivity(grouping)
+		if err != nil {
+			return nil, err
+		}
+		generalConnectivityMap1 = connectivity1.AllowedConnsCombined.nodesConnectivityToGeneralConnectivity()
+		generalConnectivityMap2 = connectivity2.AllowedConnsCombined.nodesConnectivityToGeneralConnectivity()
```

Comment:
consider separating to another function, or to have another function from the config that directly returns the `AllowedConnsCombined` of type `GeneralConnectivity`, so that a common code can be used here for both endpoints/subnets analysis.

Reply:
Done